### PR TITLE
Add initial processes documentation

### PR DESCRIPTION
**Pull request by atalyaalon**

### CODE REVIEW

**Review by zharpaz — APPROVED** [review]
Good work!
it provides high-level overview.
I think it is worthwhile to add
- The participating DB tables in each process. 
- Maybe data that is generated (Vs. copied)?
- Main data items and / or format of the resulting data

**Review by ziv17 — APPROVED** [review]
Good work!
it provides high-level overview.
I think it is worthwhile to add

    The participating DB tables in each process.
    Maybe data that is generated (Vs. copied)?
    Main data items and / or format of the resulting data

**Review by OriHoch — APPROVED** [review]
that's very helpful, thanks!

**Comment by atalyaalon** [comment]
> Good work!
> it provides high-level overview.
> I think it is worthwhile to add
> 
> ```
> The participating DB tables in each process.
> Maybe data that is generated (Vs. copied)?
> Main data items and / or format of the resulting data
> ```



> Good work!
> it provides high-level overview.
> I think it is worthwhile to add
> 
> ```
> The participating DB tables in each process.
> Maybe data that is generated (Vs. copied)?
> Main data items and / or format of the resulting data
> ```

@ziv17 @OriHoch 
Added DB tables, and future tasks I think should be on our road map
We'll need to think what's the best way t to document resulting data format / main data items, and data generated/copied.
Let me know what you think of current version

**Review by OriHoch — APPROVED** [review]
looks good, very useful

File: docs/Architecture/ARCHITECTURE_BUILDING_BLOCKS.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+ANYWAY Data
+================
+
+- CBS
+- Waze
+- OSM
+- News Flash
+- Infographics Data
+- Road Keepers
```
Comment (ziv17) [inline docs/Architecture/ARCHITECTURE_BUILDING_BLOCKS.md:9]: Maybe add where these items reside, e.g. database, s3, static files, etc.

File: docs/Architecture/CBS.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+## Current Processes
+
+### CBS: pulls data from email to s3
+- **Environment Variables**: MAILPASS, MAILUSER
+- **Relevant input**: input emails from CBS containing zip files
+- **Relevant output**: unzipped CBS files saved in S3
+- **Relevant Storage directory**: S3 -> Bucket: anyway-cbs
+- **command**: `python3 main.py scripts importemail`
```
Comment (ziv17) [inline docs/Architecture/CBS.md:8]: Maybe add where the code resides.

File: docs/Architecture/CBS.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+## Current Processes
+
+### CBS: pulls data from email to s3
```
Comment (ziv17) [inline docs/Architecture/CBS.md:3]: I did not find this step in ANYWAY Media Flow (Miro)

File: docs/Architecture/INFOGRAPHICS_DATA.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+## Current Processes
+
+## Data created in News Flash process
+- **Relevant output DB Tables**: infographics_data_cache
```
Comment (ziv17) [inline docs/Architecture/INFOGRAPHICS_DATA.md:4]: NewsFlash table should not be here?

File: docs/Architecture/NEWS_FLASH.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+## Current Processes
+
+## News Flash:
+Extracts data from source, processes it and pushes it to NewsFlash table
+In Addition - the InfographicsDataCache is filled with relevant data of the newsflash
+
+- **Environment Variables**: GOOGLE_MAPS_KEY (for google's location API)
+- **Additional environment variables for the each source**:
```
Comment (ziv17) [inline docs/Architecture/NEWS_FLASH.md:8]: Type, `the` is redundant in this line

**Review by ziv17 — APPROVED** [review]
See comments below.
Maybe we can generate a flowchart that will hold all our processes, and it will have links to these documents.

**Comment by ziv17** [comment]
> @ziv17 @OriHoch
> Added DB tables, and future tasks I think should be on our road map
> We'll need to think what's the best way t to document resulting data format / main data items, and data generated/copied.
> Let me know what you think of current version

Maybe one sentence in the flow-diagram, and more detailed data in the doc-string of the main function or module? Or on an .md file linked from the flow-diagram.

File: docs/Architecture/ARCHITECTURE_BUILDING_BLOCKS.md (hunk 1 above)
Comment (atalyaalon) [reply docs/Architecture/ARCHITECTURE_BUILDING_BLOCKS.md:9]: @ziv17 the attached documents for each logical section aren't enough?

File: docs/Architecture/CBS.md (hunk 3 above)
Comment (atalyaalon) [reply docs/Architecture/CBS.md:3]: Thanks! I modified it

File: docs/Architecture/CBS.md (hunk 2 above)
Comment (atalyaalon) [reply docs/Architecture/CBS.md:8]: @ziv17 what do you mean? In the scheduling section below it says it's in Jenkins

**Comment by atalyaalon** [comment]
> > @ziv17 @OriHoch
> > Added DB tables, and future tasks I think should be on our road map
> > We'll need to think what's the best way t to document resulting data format / main data items, and data generated/copied.
> > Let me know what you think of current version
> 
> Maybe one sentence in the flow-diagram, and more detailed data in the doc-string of the main function or module? Or on an .md file linked from the flow-diagram.

@ziv17 regarding this - any recommended tool for this?

**Comment by atalyaalon** [comment]
> See comments below.
> Maybe we can generate a flowchart that will hold all our processes, and it will have links to these documents.

@ziv17 regarding this - any recommended tool for this?

**Comment by atalyaalon** [comment]
> > @ziv17 @OriHoch
> > Added DB tables, and future tasks I think should be on our road map
> > We'll need to think what's the best way t to document resulting data format / main data items, and data generated/copied.
> > Let me know what you think of current version
> 
> Maybe one sentence in the flow-diagram, and more detailed data in the doc-string of the main function or module? Or on an .md file linked from the flow-diagram.

Great idea!

File: docs/Architecture/ARCHITECTURE_BUILDING_BLOCKS.md (hunk 1 above)
Comment (ziv17) [reply docs/Architecture/ARCHITECTURE_BUILDING_BLOCKS.md:9]: Formally yes, but someone who reads it for impression will not stop here and go to each document to find where it resides. A few words of location for each item will give a clearer high-level view.

File: docs/Architecture/CBS.md (hunk 2 above)
Comment (ziv17) [reply docs/Architecture/CBS.md:8]: I mean the python code, file or directory or group of files (e.g. infographics*)

**Comment by ziv17** [comment]
> > See comments below.
> > Maybe we can generate a flowchart that will hold all our processes, and it will have links to these documents.
> 
> @ziv17 regarding this - any recommended tool for this?

I was thinking about Miro (the one you used) or maybe [diagrams.net](https://app.diagrams.net)

**Pull request merged**